File: algorithm/2월3주차/기능개발_한도현.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> pro;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	pro = progresses;
	int size = pro.size();
	int target = 0;
	while (target != size) {
		if (pro[target] >= 100) {
			int cnt = 0;
			int to = target;
			while (to != size) {
				if (pro[to] >= 100) {
					cnt++;
					target++;
				}
				else {
					break;
				}
				to++;
			}
			answer.push_back(cnt);
		}
		for (int i = target; i < size; i++) {
			pro[i] += speeds[i];
		}
	}

	return answer;
}

File: algorithm/2월3주차/기능개발_한도현.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int idx = 0;
        int N = progresses.length;
        ArrayList<Integer> tempAns = new ArrayList<Integer>();
        while(idx<N){
            for(int i=0;i<N;i++){
                progresses[i] += speeds[i];
            }
            if(progresses[idx] >= 100){
                int cnt = 0;
                for(int i=idx;i<N;i++){
                    if(progresses[i]<100) break;
                    idx++;
                    cnt++;
                }
                tempAns.add(cnt);
            }
        }
        int[] answer = new int[tempAns.size()];
        for(int i=0;i<tempAns.size();i++){
            answer[i] = tempAns.get(i);
        }
        return answer;
    }
}
